app/controllers/main/routes.py

```diff
@@ -378,25 +379,44 @@ def getAllCourseInstructors(term=None):
     """
     This function selects all the Instructors Name and the previous courses
     """
+    showPreviewModal = request.args.get('showPreviewModal', default=False, type=bool)
+    
+    if showPreviewModal and 'courseParticipantPreview' in session:
+        courseParticipantPreview = session['courseParticipantPreview']
+    else:
+        courseParticipantPreview = []
+
+    errorFlag = session.get('errorFlag')
+    errorList = session.get('errorList')
+
     if g.current_user.isCeltsAdmin:
         setRedirectTarget(request.full_path)
         courseDict = getCourseDict()
-
         term = Term.get_or_none(Term.id == term) or g.current_term
 
         unapproved = unapprovedCourses(term)
         approved = approvedCourses(term)
         terms = selectSurroundingTerms(g.current_term)
 
+        if request.method =='POST' and "submitParticipant" in request.form:
+            pushCourseParticipantsToDatabase(session['courseParticipantPreview'])
+            courseParticipantPreviewSessionCleaner()
+            flash('File saved successfully!', 'success')
+            return redirect(url_for('main.getAllCourseInstructors'))
+      
         return render_template('/main/manageServiceLearningFaculty.html',
                                 courseInstructors = courseDict,
                                 unapprovedCourses = unapproved,
                                 approvedCourses = approved,
                                 terms = terms,
                                 term = term,
-                                CourseStatus = CourseStatus)
+                                CourseStatus = CourseStatus, 
+                                errorFlag = errorFlag,
+                                courseParticipantPreview= courseParticipantPreview,
+                                errorList = errorList
```

Comment:
errorFlag and errorList are too generic, considering there are other things going on in this template